### PR TITLE
Make archive validation error messages friendlier

### PR DESCRIPTION
**Pull request by ubergesundheit**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**
User experience improvement

- **What is the current behavior?**
Trying to install packages with `core install <fqdn>` is not helpful when:

- Index file has incorrect size for a package file
- Index file has no checksum for a package file

* **What is the new behavior?**
This PR improves the errors printed when the package file contains incorrect values

- **Does this PR introduce a breaking change, and is
[titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
no

* **Other information**:
Tested with index file https://raw.githubusercontent.com/sensebox/senseBoxMCU-core/2feebd752e3b86e5b71bd2c4addfd3188f349885/package_sensebox_index.json containing a wrong size and no checksum for the latest package.

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

**Comment by ubergesundheit** [comment]
Seems I should've run the tests locally. I'll fix everything later

**Comment by ubergesundheit** [comment]
@cmaglie @silvanocerza sorry for pinging you, but I could not find any information who to tag in PR and I think this PR adds useful changes.

Please let me know if I need to change something. Thanks :)

**Comment by ubidefeo** [comment]
hey @ubergesundheit 
we have a few days of heavy workload ahead but I'll make sure this ends up being looked at next week at the latest

**Review by silvanocerza — APPROVED** [review]
Nice! Thanks for the contribution, better error handling is always welcome. :)

File: arduino/resources/checksums.go [hunk 1]
```diff
@@ -163,5 +175,9 @@ func CheckDirChecksum(root string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	return file.Checksum == checksum, nil
+	if file.Checksum != checksum {
+		return false, fmt.Errorf("Checksum differs from checksum in package.json")
```
Comment (ubergesundheit) [inline arduino/resources/checksums.go:179]: Please wait before merge. I think `package.json` is not correct here.

```suggestion
		return false, fmt.Errorf("Checksum differs from checksum in package index")
```

Comment (silvanocerza) [reply arduino/resources/checksums.go:179]: Mmmh, I think it's correct as it is. You compare the checksum read from `package.json` and the one you calculate given the `root` so the error makes sense.

Comment (ubergesundheit) [reply arduino/resources/checksums.go:179]: Ok :)

**Pull request merged**